gunicorn/sock.py

```diff
@@ -88,6 +93,18 @@ def set_options(self, sock, bound=False):
         sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
         return super().set_options(sock, bound=bound)
 
+    def get_backlog(self):
+        if self.sock and PLATFORM == "linux":
+            # tcp_info struct from include/uapi/linux/tcp.h
+            fmt = 'B'*8+'I'*24
+            try:
+                tcp_info_struct = self.sock.getsockopt(socket.IPPROTO_TCP,
+                                                      socket.TCP_INFO, 104)
+                # 12 is tcpi_unacked
+                return struct.unpack(fmt, tcp_info_struct)[12]
+            except AttributeError:
+                pass
+        return 0
```

Comment:
Any thoughts on returning None here, so users can distinguish between "unsupported" and "the backlog is empty"?

Reply:
Makes sense, I've made the changes.